src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -149,11 +149,19 @@ private LeaseContext CombinedAcquire(HttpContext context)
                 return new LeaseContext() { GlobalRejected = false, Lease = endpointLease };
             }
         }
-        catch (Exception)
+        catch (Exception ex)
         {
             endpointLease?.Dispose();
             globalLease?.Dispose();
-            throw;
+            if (ex is TaskCanceledException)
+            {
+                RateLimiterLog.TaskCanceled(_logger);
+                return new LeaseContext() { Lease = new DefaultAcquiredLease() };
```

Comment:
Could this lease could be shared?
```suggestion
                return new LeaseContext() { Lease = DefaultAcquiredLease.Instance };
```

Reply:
I don't think we'll end up exposing this to users since I agree we shouldn't call `OnRejected` in this case. So maybe we can make this super easy and return a `LeaseContext?` instead of a `LeaseContext` and special case `null` instead. Then we don't need this class at all.

Reply:
`LeaseContext` is an internal struct, so we could make it nullable. We could also add a flag like `LeaseContext.GlobalRejected` instead of using `Nullable<T>`. Maybe the flags could even be stuffed into a flag enum.